plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/utils/CodeTransformFileUtils.kt

```diff
@@ -123,11 +138,55 @@ fun unzipFile(zipFilePath: Path, destDir: Path): Boolean {
 
 fun parseXmlDependenciesReport(pathToXmlDependency: Path): DependencyUpdatesReport {
     val reportFile = pathToXmlDependency.toFile()
-    val xmlMapper = XmlMapper()
-    val report = xmlMapper.readValue(reportFile, DependencyUpdatesReport::class.java)
+    val report = XML_MAPPER.readValue(reportFile, DependencyUpdatesReport::class.java)
     return report
 }
 
+fun validateSctMetadata(sctFile: File?): SqlMetadataValidationResult {
+    if (sctFile == null) {
+        return SqlMetadataValidationResult(false, message("codemodernizer.chat.message.validation.error.missing_sct_file"))
+    }
+
+    val sctMetadata: SctMetadata
+    try {
+        sctMetadata = XML_MAPPER.readValue<SctMetadata>(sctFile)
+    } catch (e: Exception) {
+        getLogger<CodeTransformChatController>().error { "Error parsing .sct metadata file; invalid XML encountered. $e" }
```

Comment:
our linter is broken and not catching this for some reason

```suggestion
        getLogger<CodeTransformChatController>(e).error { "Error parsing .sct metadata file; invalid XML encountered." }
```

Reply:
When I tried this, got errors prompting me to make changes in `LogUtils.kt`, wasn't sure if I should do that, ideally that can be a follow up, for now I changed it to just log the localizedMessage rather than dumping the entire Error object